app/views/components/_autocomplete.html.erb

```diff
@@ -0,0 +1,68 @@
+<%
+  id ||= "autocomplete-#{SecureRandom.hex(4)}"
```

Comment:
If this auto complete is now being used in two applications, should it instead be made into a component in govuk_publishing_components?

cc @DilwoarH

Reply:
This is already in there (https://govuk-publishing-components.herokuapp.com/component-guide/accessible_autocomplete) but I think this version has additional features which the gem version doesn't. I will have a look to see if we can move it to the gem.

Reply:
Absolutely! I think it's worth doing the work here for now while we figure out what we need from the component (for example, I said in the commit messages I'd like to add an `error_message` parameter rather than use the more verbose `error_items` parameter).

Once we've added to collections-publisher, work on a joined up plan to shift it out to govuk_publishing_components and remove it from Content Publisher and Collections Publisher.

There's talk about retiring the existing [autocomplete component](https://govuk-publishing-components.herokuapp.com/component-guide/accessible_autocomplete) in favour of the one in Content Publisher (which I think is having a11y improvements made to it at the moment...?). We'd need to liaise with @alex-ju on that.

Reply:
Yeah, we should chat about how to share this bit of code efficiently - on which we constantly work it seems.

Reply:
Are there any tests for this component? Can you add some?

Reply:
What kind of tests? Do you mean Ruby integration tests that ensure things like the presence of `error_items` actually renders errors...? (NB, there were no tests in Content Publisher to copy over).

I think JS unit tests would be ideal, but am conscious that it would be scope creep to write tests for features we're not using (such as multi-option select) - but I wouldn't want to start removing features either, for [reasons outlined in the other thread](https://github.com/alphagov/collections-publisher/pull/664#discussion_r309178694). I'm happy to add some tests (once we've clarified which!) but perhaps that work can be more thoroughly done when it's extracted out to govuk_publishing_components.

Reply:
Have added tests later on in this PR, and have stripped down the component so it only has the features we need, as requested 👍 

Will arrange a chat with Alex & Ollie & Dilwoar in the next week or two to see if we can decide where this component should be moved to, as it doesn't make sense for it to be duplicated across repositories.